From 5204eaac0c0f6277fdf5d9f7cc40a327216b9107 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:14:36 -0400
Subject: [ethernet] be2net: remove unncessary gotos

Message-id: <1410192881-30702-74-git-send-email-ivecera@redhat.com>
Patchwork-id: 92452
O-Subject: [RHEL7.1 PATCH 73/78] be2net: remove unncessary gotos
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit 3fb8cb801a9393888c09bea38d3e9ca360221b66
Author: Kalesh AP <kalesh.purayil@emulex.com>
Date:   Tue Sep 2 09:56:52 2014 +0530

    be2net: remove unncessary gotos

    In cases where there is no extra code to handle an error, this patch replaces
    gotos with a direct return statement.

    Signed-off-by: Kalesh AP <kalesh.purayil@emulex.com>
    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index 96db825..927410d 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -4086,18 +4086,15 @@ static int lancer_fw_download(struct be_adapter *adapter,
 
  if (!IS_ALIGNED(fw->size, sizeof(u32))) {
   dev_err(dev, "FW image size should be multiple of 4\n");
-  status = -EINVAL;
-  goto lancer_fw_exit;
+  return -EINVAL;
  }
 
  flash_cmd.size = sizeof(struct lancer_cmd_req_write_object)
     + LANCER_FW_DOWNLOAD_CHUNK;
  flash_cmd.va = dma_alloc_coherent(dev, flash_cmd.size,
        &flash_cmd.dma, GFP_KERNEL);
- if (!flash_cmd.va) {
-  status = -ENOMEM;
-  goto lancer_fw_exit;
- }
+ if (!flash_cmd.va)
+  return -ENOMEM;
 
  dest_image_ptr = flash_cmd.va +
     sizeof(struct lancer_cmd_req_write_object);
@@ -4135,7 +4132,7 @@ static int lancer_fw_download(struct be_adapter *adapter,
  dma_free_coherent(dev, flash_cmd.size, flash_cmd.va, flash_cmd.dma);
  if (status) {
   dev_err(dev, "Firmware load error\n");
-  goto lancer_fw_exit;
+  return be_cmd_status(status);
  }
 
  dev_info(dev, "Firmware flashed successfully\n");
@@ -4147,13 +4144,12 @@ static int lancer_fw_download(struct be_adapter *adapter,
   if (status) {
    dev_err(dev, "Adapter busy, could not reset FW\n");
    dev_err(dev, "Reboot server to activate new FW\n");
-   goto lancer_fw_exit;
   }
  } else if (change_status != LANCER_NO_RESET_NEEDED) {
   dev_info(dev, "Reboot server to activate new FW\n");
  }
-lancer_fw_exit:
- return status;
+
+ return 0;
 }
 
 #define UFI_TYPE2  2
-- 
1.7.1